cmd/authorization/main.go

```diff
@@ -74,9 +93,119 @@ func uppercaseHandler(w http.ResponseWriter, req *http.Request) {
 	return
 }
 
+func createUser(db *sql.DB, login, password_hash string) error {
+	createUserStmt := `INSERT INTO users VALUES($1, $2)`
+	_, err := db.Exec(createUserStmt, login, password_hash)
+
+	if err != nil {
+		log.Println("Error create user " + err.Error())
+		return err
+	}
+	return err
+}
+func readUser(db *sql.DB, login string) (User, error) {
+	var user User
+	readUserStmt := `SELECT * FROM users WHERE login = $1`
+
+	rows, err := db.Query(readUserStmt, login)
+	if err != nil {
+		log.Println("Error get user " + err.Error())
+		return user, err
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		err = rows.Scan(&user)
+		if err != nil {
+			log.Println("Error get data from row " + err.Error())
+			return user, err
```

Comment:
You lost an error context here, it's better to use `fmt.Errorf` here. E.g.:
```
return user, fmt.Errorf("unable to read user: %w", err)
```
See docs https://pkg.go.dev/fmt#Errorf

Reply:
okay